docs/howto/manage-charms.md

```diff
@@ -209,23 +197,34 @@ juju deploy some-bundle --map-machines=existing,3=4,4=5
 
 ````
 
-Depending on the cloud substrate that your controller is running on, the above command will allocate a machine (physical,  virtual, LXD container) or a Kubernetes pod and then proceed to deploy the contents of the charm / bundle.
+````{dropdown} Tips for troubleshooting - machines
+Deploy on machines consists of the following steps: Provision resources/a machine M from the relevant cloud, via cloud-init maybe network config, download the `jujud` binaries from the controller, start `jujud`.
 
-```{note}
+For failure at any point, retry the `deploy` command with the `--debug` and `--verbose` flags:
 
-Depending on your use case, you may alternatively opt to provision a set of machines in advance via the `juju add-machine` command.
+```text
+juju deploy <charm> --debug --verbose
+```
 
-In this case, when running the above `juju deploy` command, Juju will detect that the model contains machines with no applications assigned to them and automatically deploy the application to one of those machines instead of spinning up a new machine.
+If it still fails,  connect to the machine and examine the logs.
 
+```{ibnote}
+See more: {ref}`manage-logs`, {ref}`troubleshoot-your-deployment`
 ```
 
-The command also allows you to add another argument to specify a custom name (alias) for your deployed application (charms only). You can also take advantage of the rich set of flags to specify a charm channel or revision, a machine base, a machine constraint (e.g., availability zone), the number of application units you want (clusterised), a space binding, a placement directive (e.g., to deploy to a LXD container), a specific storage instance, a specific machine, etc., and even to trust the application with the current credential -- in case the application requires access to the backing cloud in order to fulfil its purpose (e.g., stojrage-related tasks).
+If Juju fails to provision a subset of machines for some reason (e.g., machine quota limits on the cloud provider), you can also use the {ref}`command-juju-retry-provisioning` command to retry the provisioning of specific machine numbers.
```

Comment:
This is tricky, it only works for one specific error on one cloud afaik. Trying to remember the details. And does not work for containers. I've never seen it work in 8+ years.

Reply:
I'll comment it out with a note to follow up and your comment. Thanks!